kie-server-parent/kie-server-controller/kie-server-controller-impl/src/test/java/org/kie/server/controller/impl/storage/FileBasedKieServerTemplateStorageTest.java

```diff
@@ -0,0 +1,279 @@
+ * limitations under the License.
+*/
+package org.kie.server.controller.impl.storage;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.kie.server.api.model.KieContainerStatus;
+import org.kie.server.api.model.KieScannerStatus;
+import org.kie.server.api.model.ReleaseId;
+import org.kie.server.controller.api.model.runtime.Container;
+import org.kie.server.controller.api.model.spec.Capability;
+import org.kie.server.controller.api.model.spec.ContainerConfig;
+import org.kie.server.controller.api.model.spec.ContainerSpec;
+import org.kie.server.controller.api.model.spec.ProcessConfig;
+import org.kie.server.controller.api.model.spec.RuleConfig;
+import org.kie.server.controller.api.model.spec.ServerTemplate;
+import org.kie.server.controller.api.model.spec.ServerTemplateKey;
+import org.kie.server.controller.api.service.RuntimeManagementService;
+import org.kie.server.controller.api.service.SpecManagementService;
+import org.kie.server.controller.impl.KieServerInstanceManager;
+import org.kie.server.controller.impl.service.RuntimeManagementServiceImpl;
+import org.kie.server.controller.impl.service.SpecManagementServiceImpl;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Maps;
+
+public class FileBasedKieServerTemplateStorageTest {
+    private static final Logger logger = LoggerFactory.getLogger(FileBasedKieServerTemplateStorageTest.class);
+    private static final String TEST_SERVER_TEMPLATE_FILENAME = System.getProperty("java.io.tmpdir")
+    		+ System.getProperty("file.separator")
+    		+ "test_template.xml";
+    private static SpecManagementService specManageService;
+    private static RuntimeManagementService runtimeManagementService;
+    private static KieServerInstanceManager kieServerInstanceManager;
+    
+    private static Map<ServerTemplateKey, ServerTemplate> templateMap;
+    private FileBasedKieServerTemplateStorage storage;
+    
+    /**
+     * Method that creates a server template instance and saves it in the specManageService
+     * @param templateName The name of the template being created
+     * @param templateCount The expected number of templates saved in the specManageService, including the one being created 
+     * @return A new server template instance
+     */
+    private static ServerTemplate createServerTemplateWithContainer(String templateName, int templateCount) {
```

Comment:
The version of this method, in FileBasedKieServerTemplateStorageTest, takes two parameters that are necessary for the purpose of testing multiple template situations. The version in AbstractServiceImplTest only expects the creation of a single server template (see line 62 of AbstractServiceImplTest). If necessary, I can break apart the method in AbstractServiceImplTest and create multiple methods that do the work that isn't dependent on the server template name and the number of templates that are expected to be in the specManagementService. I'm not sure that this is worth that level of effort, though.